airbyte-cdk/python/unit_tests/sources/file_based/in_memory_files_source.py

```diff
@@ -209,3 +209,23 @@ def _make_file_contents(self, file_name: str) -> bytes:
             file_writer.flush()
             fp.seek(0)
             return fp.read()
+
+
+class TemporaryExcelFilesStreamReader(InMemoryFilesStreamReader):
+    """
+    A file reader that writes RemoteFiles to a temporary file and then reads them back.
+    """
+
+    def open_file(self, file: RemoteFile, mode: FileReadMode, encoding: Optional[str], logger: logging.Logger) -> IOBase:
+        return io.BytesIO(self._make_file_contents(file.uri))
+
+    def _make_file_contents(self, file_name: str) -> bytes:
+        contents = self.files[file_name]["contents"]
+        df = pd.DataFrame(contents)
+
+        with io.BytesIO() as fp:
+            writer = pd.ExcelWriter(fp, engine='xlsxwriter')
```

Comment:
Why do we use a different engine than [here](https://github.com/airbytehq/airbyte/pull/43346/files#diff-aa4686a61ca7e6f6525983b4aa54e329a0eea38502225700d8b11398e6cbc8b0R56)?

Reply:
in the code we use a reading engine, but this is a writing engine